packages/components/color-picker/src/index.vue

```diff
@@ -127,263 +125,219 @@ import {
-    const svPanel = ref<InstanceType<typeof SvPanel>>()
-    const alpha = ref<InstanceType<typeof AlphaSlider>>()
-    const popper = ref(null)
-    // active-change is used to prevent modelValue changes from triggering.
-    let shouldActiveChange = true
-    // data
-    const color = reactive(
-      new Color({
-        enableAlpha: props.showAlpha,
-        format: props.colorFormat || '',
-        value: props.modelValue,
-      })
-    )
-    type ColorType = typeof color
-    const showPicker = ref(false)
-    const showPanelColor = ref(false)
-    const customInput = ref('')
-    // computed
-    const displayedColor = computed(() => {
-      if (!props.modelValue && !showPanelColor.value) {
-        return 'transparent'
-      }
-      return displayedRgb(color, props.showAlpha)
-    })
-    const colorSize = useSize()
-    const colorDisabled = computed(() => {
-      return !!(props.disabled || form?.disabled)
-    })
+const { t } = useLocale()
+const ns = useNamespace('color')
+const { form, formItem } = useFormItem()
 
-    const currentColor = computed(() => {
-      return !props.modelValue && !showPanelColor.value ? '' : color.value
-    })
-    const buttonAriaLabel = computed<string | undefined>(() => {
-      return !isLabeledByFormItem.value
-        ? props.label || t('el.colorpicker.defaultLabel')
-        : undefined
-    })
-    const buttonAriaLabelledby = computed<string | undefined>(() => {
-      return isLabeledByFormItem.value ? formItem?.labelId : undefined
-    })
-    // watch
-    watch(
-      () => props.modelValue,
-      (newVal) => {
-        if (!newVal) {
-          showPanelColor.value = false
-        } else if (newVal && newVal !== color.value) {
-          shouldActiveChange = false
-          color.fromString(newVal)
-        }
-      }
-    )
-    watch(
-      () => currentColor.value,
-      (val) => {
-        customInput.value = val
-        shouldActiveChange && emit('active-change', val)
-        shouldActiveChange = true
-      }
-    )
-
-    watch(
-      () => color.value,
-      () => {
-        if (!props.modelValue && !showPanelColor.value) {
-          showPanelColor.value = true
-        }
-      }
-    )
+const { inputId: buttonId, isLabeledByFormItem } = useFormItemInputId(props, {
+  formItemContext: formItem,
+})
 
-    // methods
-    function displayedRgb(color: ColorType, showAlpha: boolean) {
-      if (!(color instanceof Color)) {
-        throw new TypeError('color should be instance of _color Class')
-      }
+const hue = ref<InstanceType<typeof HueSlider>>()
+const svPanel = ref<InstanceType<typeof SvPanel>>()
+const alpha = ref<InstanceType<typeof AlphaSlider>>()
+const popper = ref(null)
+// active-change is used to prevent modelValue changes from triggering.
+let shouldActiveChange = true
+// data
+const color = reactive(
+  new Color({
+    enableAlpha: props.showAlpha,
+    format: props.colorFormat || '',
+    value: props.modelValue,
+  })
+) as Color
```

Comment:
```suggestion
const color = reactive<Color>(
  new Color({
    enableAlpha: props.showAlpha,
    format: props.colorFormat || '',
    value: props.modelValue,
  })
)
```

Reply:
resolved🙌

Reply:
i think it should use as do type assert,becasue reactive<Color> !== Color